packages/server/src/api/muster/muster.controller.ts

```diff
@@ -231,30 +233,76 @@ class MusterController {
   ) {
-    const outValue = { musterComplianceRate: 0 };
-    const reportDate = dateFromString(req.query.timestamp);
-    if (!reportDate) {
-      throw new BadRequestError('Missing reportDate.');
+    const outValue = { musterComplianceRate: 0, isoDate: req.query.isoDate };
+    outValue.musterComplianceRate = await getMusterComplianceByDate(req.params.orgId, req.params.unitName, req.query.isoDate);
+    res.json(outValue);
+  }
+
+  /**
+   * This method returns the normalized rate (0 - 1.0) of compliance on a given
+   * date range for a given unit, against all of the unit's muster configs.
+   */
+  async getMusterComplianceByDateRange(
+    req: ApiRequest<{ orgId: number; unitName: string }, null, GetMusterComplianceByDateRangeQuery>,
+    res: Response<GetMusterComplianceByDateRangeResponse>,
+  ) {
+    const out: GetMusterComplianceByDateRangeResponse = { musterComplianceRates: [] };
+    const range: string[] = req.query.isoDateRange.split('/');
+    if (range.length !== 2) {
+      throw new BadRequestError('Invalid ISO date range.');
+    } else if (range[0] === range[1]) {
+      const compliance = await getMusterComplianceByDate(req.params.orgId, req.params.unitName, range[0]);
+      out.musterComplianceRates.push({ musterComplianceRate: compliance, isoDate: range[0] });
+    } else {
+      let fromDate: moment.Moment = moment(range[0]);
+      const toDate: moment.Moment = moment(range[1]);
+      while (fromDate <= toDate) {
+        const compliance = await getMusterComplianceByDate(req.params.orgId, req.params.unitName, fromDate.toISOString());
+        out.musterComplianceRates.push({ musterComplianceRate: compliance, isoDate: fromDate.toISOString() });
+        fromDate = fromDate.add(1, 'day');
+      }
     }
+    res.json(out);
+  }
+
+}
+
+/**
+ * This helper method returns the normalized rate (0 - 1.0) of compliance on a given
+ * date for a given unit, against all of the unit's muster configs.
+ * @param orgId - the organization id
+ * @param unitName - the roster the users belong to
+ * @param isoDate - the date in format YYYY-MM-DD
+ */
+async function getMusterComplianceByDate(orgId: number, unitName: string, isoDate: string) {
+  let outValue: number = 0;
+  let reportDate: Date | undefined;
+  try {
+    reportDate = dateFromString(isoDate);
+  } catch (DateParseError) {
+    throw new BadRequestError('Invalid ISO date.');
+  }
+
+  if (reportDate) {
     // eslint-disable-next-line no-bitwise
     const dayBit = 1 << reportDate.getDay();
 
-    const usersOnRoster = await getUsersOnRosterByDate(req.params.orgId, req.params.unitName, req.query.timestamp);
+    const usersOnRoster = await getUsersOnRosterByDate(orgId, unitName, isoDate);
     const users = usersOnRoster.users;
     const userCount = users.length;
     let musterAvg: number = 0;
 
     if (usersOnRoster.musterConfig.length === 0 || userCount === 0) {
-      res.json(outValue);
-      return;
+      return outValue;
```

Comment:
You could simplify it by returning zero here and when **reportDate** is **falsy** (fail fast). By declaring **outValue = 0** at the top and using it in the middle and very bottom of the function makes one look for what the value is at these locations which is not necessary.